service/storage_proxy.cc

```diff
@@ -721,8 +742,54 @@ class storage_proxy::remote {
         }
         shared_ptr<storage_proxy> p = _sp.shared_from_this();
         auto cmd = make_lw_shared<query::read_command>(std::move(cmd1));
+
         auto src_ip = src_addr.addr;
         auto timeout = t ? *t : db::no_timeout;
+
+        auto f_r = co_await coroutine::as_future([&]() -> future<bool> {
+            // Check if we have a forward query
+            if (!cmd->slice.is_reversed())
+                return ::make_ready_future<bool>(false);
+
+            // We deal with a reversed query. Check if read command is coming from an old node.
+            // Table schema is then passed with the read command.
+            auto schema = p->local_db().find_schema(cmd->cf_id);
+            if (cmd->schema_version == schema->version()) {
+                (void)schema->get_reversed();
+                return ::make_ready_future<bool>(true);
+            }
+
+            // We deal with a reversed query. Check if read command is coming from an new node.
+            // Reversed table schema is then passed with the read command.
```

Comment:
Same here: "If coordinator is a new node, it will send the reversed table schema in the read command".